ts/features/walletV3/payment/screens/WalletPaymentPickMethodScreen.tsx

```diff
@@ -44,51 +83,238 @@ const WalletPaymentPickMethodScreen = () => {
+              methods =>
+                (methods.map(mapSavedToRadioItem) ?? []) as Array<
+                  RadioItem<string>
+                >
+            )
+          ),
+    [isLoading, userWalletsPots]
+  );
+
+  const notSavedMethodsListItems = useMemo(
+    () =>
+      isLoading
+        ? loadingRadios
+        : pipe(
+            paymentMethodsPot,
+            pot.toOption,
+            O.fold(
+              () => [],
+              methods =>
+                methods.map(item => {
+                  const radio = mapGenericToRadioItem(item, paymentAmount);
+                  if (radio.disabled === true) {
+                    setShouldShowWarningBanner(true);
+                  }
+                  return radio;
+                })
+            )
+          ),
+    [isLoading, paymentMethodsPot, paymentAmount]
+  );
+
+  useHeaderSecondLevel({
+    title: "",
+    backAccessibilityLabel: I18n.t("global.buttons.back"),
+    goBack: navigation.goBack,
+    contextualHelp: emptyContextualHelp,
+    faqCategories: ["payment"],
+    supportRequest: true
+  });
+
+  // -------------------------- RENDER --------------------------
+
+  const alertRef = React.useRef<View>(null);
 
   return (
-    <BaseScreenComponent goBack={true}>
-      <GradientScrollView
-        primaryActionProps={{
-          label: "Continua",
-          accessibilityLabel: "Continua",
-          onPress: handleContinue,
-          disabled: isLoading || !canContinue,
-          loading: isLoading
-        }}
-      >
-        <DebugPrettyPrint title="paymentMethodsPot" data={paymentMethodsPot} />
-        <VSpacer size={16} />
-        <DebugPrettyPrint title="userWalletsPots" data={userWalletsPots} />
-      </GradientScrollView>
-    </BaseScreenComponent>
+    <GradientScrollView
+      primaryActionProps={{
+        label: I18n.t("global.buttons.continue"),
+        accessibilityLabel: I18n.t("global.buttons.continue"),
+        onPress: handleContinue,
+        disabled: isLoading || !canContinue,
+        loading: isLoading
+      }}
+    >
+      <H2>{I18n.t("wallet.methodSelection.header")}</H2>
+      <VSpacer size={16} />
+      {shouldShowWarningBanner && (
+        <Alert
+          content={I18n.t("wallet.methodSelection.alert.body")}
+          variant="warning"
+          viewRef={alertRef}
+          onPress={() => null}
+          action={I18n.t("wallet.methodSelection.alert.cta")}
+        />
+      )}
+      <ListItemHeader label={I18n.t("wallet.methodSelection.yourMethods")} />
+
+      <RadioGroup<string>
+        selectedItem={selectedMethod?.walletId}
+        items={savedMethodsListItems}
+        onPress={handleSelectSavedMethod}
+      />
+
+      <ListItemHeader label={I18n.t("wallet.methodSelection.otherMethods")} />
+
+      <RadioGroup<string>
+        items={notSavedMethodsListItems}
+        selectedItem={selectedMethod?.methodId}
+        onPress={handleSelectNotSavedMethod}
+      />
```

Comment:
Loading state is an UI state, we should not mutate the data based on the isLoading property. I suggest to use the property directly in the render.

```suggestion
      <RadioGroup<string>
        selectedItem={selectedMethod?.walletId}
        items={isLoading ? loadingRadios : savedMethodsListItems}
        onPress={handleSelectSavedMethod}
      />

      <ListItemHeader label={I18n.t("wallet.methodSelection.otherMethods")} />

      <RadioGroup<string>
        items={isLoading ? loadingRadios : genericMethodsListItems}
        selectedItem={selectedMethod?.methodId}
        onPress={handleSelectNotSavedMethod}
      />
```